### ISSUE TITLE
Enable/disable/list repos using redhat_subscription module

### ISSUE THREAD

**Issue by ghost** [ansible/ansible-modules-core#3300]
##### Issue Type:
- Feature Idea
##### Plugin Name:

redhat_subscription
##### Ansible Version:

N/A
##### Ansible Configuration:

N/A
##### Environment:

N/A
##### Summary:

redhat_subscription module does not contains options to enable/disable/list repos.
When connecting a server to the rhn or satellite and assign a subscription without activation keys, its necessary to enable or disable repos.
##### Steps To Reproduce:

subscription-manager command provides those options:
- Enable: subscription-manager repos --enable repo [--enable repo] [--enable regexp]
- Disable: subscription-manager repos --disable repo [--disable repo] [--disable regexpr]
- List all: subscription-manager repos --list
- List enabled: subscription-manager repos --list-enabled
- List disabled: subscription-manager repos --list-disabled

**Comment by jfenal-zz** [comment]
As for needs:
- get list of repos currently enabled on the machine
- enable if not yet enabled a repos or a list of repos (in one command, as subscription-manager is slooooow)
- same for disabling a list of repos, with special '*' to disable all repos (and build on a clean state)

**Comment by samdoran** [comment]
I would like this functionality as well. I'm wondering about the best way to go about implementing it.

Would it make sense to add this to the existing `redhat_subscription` module given the `state:` parameter would then be ambiguous, i.e., is the _repo_ enabled ( `repos --enable` vs `repos --disable`) or is the _subscription_ enabled (`register` vs. `unregister`)? Adding a `repo_state` parameter would remove the amibiguity, but seems a little gross.

Or would it be better to split this out into a separate `redhat_repository` module so it's clear exactly what is being enabled/disabled ([singular name](https://github.com/ansible/proposals/issues/10) to make @abadger happy)?

**Comment by minusdavid** [comment]
Since the modules "rhn_channel" and "rhn_register" already exist, I think it would be reasonable and consistent to create a new module called "redhat_repository" to go alongside "redhat_subscription". 

By the way, I'd also like this functionality. I wonder how people are managing their repos using subscription-manager in Ansible at the moment. Using the command/shell modules I'm guessing?

**Comment by minusdavid** [comment]
As a short-term measure, I'm writing a custom "site_facts" module which will parse redhat.repo, and then I can use that with some conditionals to do "command: subscription-manager repos...".

It's not really that different from the concept behind http://docs.ansible.com/ansible/include_vars_module.html I imagine... except instead of YAML/JSON it's the INI type format that Python uses with ConfigParser.

I'm going to look at it again on Monday, and if I can make it nice and clean, I'll post it on my own Github account for anyone else who wants to use it.

**Comment by jfenal-zz** [comment]
I've started writing such a module, but was interrupted before I could finish a first version I could test.
Have a look at : https://github.com/jfenal/ansible-modules-jfenal
The logic is mostly there, BUT not tested at all, no JSON output, no error handling. It clearly lacks some more code to be functional, but close.

Please note my disclaimer : my first ansible module, and moreover, my first attempt to write (with lots of copy/pasting) python code. Cave canem :)

**Comment by samdoran** [comment]
I have a [role](https://galaxy.ansible.com/samdoran/redhat-subscription/) that's using `command` to manage subscriptions and repos.

**Comment by minusdavid** [comment]
Kudos for your work, guys!

@samdoran I take it that's not idempotent, so you use tags to just run it once during initial setup?

@jfenal Looks ambitious! I opted for a local custom module, which you can see at this [Gist](https://gist.github.com/minusdavid/2774a8d6df5ded46b1b63e05f214da01)

**Comment by minusdavid** [comment]
Btw, it works well enough. I like the idea of a core redhat_repository module best, but this will certainly do the job for now.

**Comment by minusdavid** [comment]
Just realized that you both appear to work at Red Hat.

I'm super curious about if/how RH is developing Ansible to help manage RHEL...

I've only been using RHEL for a couple of weeks, but I've used Ansible for a couple years. So far so good using it to setup RHEL servers. The only real sticking point so far has been this issue. Well, and this [one](https://github.com/ansible/ansible/issues/16006), but perhaps this comment section isn't the best place for discussing such things.

**Comment by detiber** [comment]
@barnabycourt not sure if you've seen this bug

**Comment by ansibot** [comment]
@barnabycourt, @alikins, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by jfenal-zz** [comment]
https://github.com/ansible/ansible-modules-extras/pull/3559

**Comment by ansibot** [comment]
This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by jfenal-zz** [comment]
For followers of this thread, please have a look at https://github.com/jfenal/ansible-modules-jfenal, all feedbacks are welcome before I submit it to ansible core.

**Comment by detiber** [comment]
@barnabycourt might be of interest to you.

**Comment by rvalente** [comment]
This issue was moved to ansible/ansible#28159
